src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -184,17 +184,17 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
 
         private async Task UninstallAsync(IRPackageViewModel package) {
             _coreShell.AssertIsOnMainThread();
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                IsLoading = true;
-            }
+            IsLoading = true;
```

Comment:
So code gets confused when something didnt happen. For example, a dependent package that user tries to uninstall won't get unloaded or uninstalled. So similarly to loading where sequence is always available -> installed -> loaded, uninstall should go in reverse. Such as, first try and unload the package and then check if it was really unloaded. If it was, then proceed with uninstall otherwise bail. I changed code to this extent and the results appears to be more predictable.